test/cross-chain/cross-chain.js

```diff
@@ -148,4 +457,992 @@ contract("Cross-chain", () => {
+
+      // And bridging again doesn't work
+      p = bridgeMonitor.getPromiseForNextBridgedTransaction();
+      tx = await foreignColonyNetwork.bridgeSkillIfNotMiningChain(foreignSkillCount);
+      await tx.wait();
+      await p;
+
+      const pendingAddition = await homeColonyNetwork.getPendingSkillAddition(foreignChainId, foreignSkillCount);
+      expect(pendingAddition.toHexString()).to.equal("0x00");
+
+      const homeSkillCount = await homeColonyNetwork.getBridgedSkillCounts(foreignChainId);
+      expect(homeSkillCount.toHexString()).to.equal(foreignSkillCount.toHexString());
+    });
+
+    it("can't bridge a skill that doesn't exist", async () => {
+      const skillCount = await foreignColonyNetwork.getSkillCount();
+      const nonExistentSkillId = skillCount.add(10000000);
+      const tx = await foreignColonyNetwork.bridgeSkillIfNotMiningChain(nonExistentSkillId, {
+        gasLimit: 1000000,
+      });
+      await checkErrorRevertEthers(tx.wait(), "colony-invalid-skill-id");
+    });
+
+    it("if bridge is broken, bridging skill transaction doesn't revert (allowing e.g. domains to be created)", async () => {
+      let tx = await foreignBridge.setBridgeEnabled(false);
+      await tx.wait();
+      const skillCount = await foreignColonyNetwork.getSkillCount();
+
+      tx = await foreignColonyNetwork.bridgeSkillIfNotMiningChain(skillCount, {
+        gasLimit: 1000000,
+      });
+      let receipt = await tx.wait();
+      expect(receipt.status).to.equal(1);
+
+      tx = await foreignColony["addDomain(uint256,uint256,uint256)"](1, UINT256_MAX_ETHERS, 1);
+      receipt = await tx.wait();
+
+      let events = receipt.logs.map(function (log) {
+        try {
+          return foreignColonyNetwork.interface.parseLog(log);
+        } catch (e) {
+          // Return nothing
+        }
+        return null;
+      });
+      events = events.filter((x) => x != null && x.eventFragment.name === "SkillCreationStored");
+      expect(events.length).to.equal(1);
+      const event = events[0];
+      expect(event.args[0].toString()).to.equal(skillCount.add(1).toString());
+    });
+
+    it("colony root local skill structures end up the same on both chains", async () => {
+      const homeColonyRootLocalSkillId = await homeColony.getRootLocalSkill();
+      let homeColonyRootLocalSkill = await homeColonyNetwork.getSkill(homeColonyRootLocalSkillId);
+
+      const foreignColonyRootLocalSkillId = await foreignColony.getRootLocalSkill();
+      let foreignColonyRootLocalSkill = await foreignColonyNetwork.getSkill(foreignColonyRootLocalSkillId);
+
+      expect(homeColonyRootLocalSkill.nParents.toString()).to.equal(foreignColonyRootLocalSkill.nParents.toString());
+      expect(homeColonyRootLocalSkill.nChildren.toString()).to.equal(foreignColonyRootLocalSkill.nChildren.toString());
+
+      let tx = await homeColony.addLocalSkill();
+      await tx.wait();
```

Comment:
Is this await necessary?

Reply:
Yes, as otherwise there will be a race condition for `getPromiseForNextBridgedTransaction` - if the bridge is slow, the next bridged transaction the promise sees could be this one, if this line is not present.